src/Collapse.js

```diff
@@ -190,17 +100,118 @@ Collapse.propTypes = {
   /**
    * ARIA role of collapsible element
    */
-  role: React.PropTypes.string
+  role: React.PropTypes.string,
 };
 
-Collapse.defaultProps = {
+const defaultProps = {
   in: false,
   timeout: 300,
   unmountOnExit: false,
   transitionAppear: false,
 
   dimension: 'height',
-  getDimensionValue
+  getDimensionValue,
 };
 
+class Collapse extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.handleEnter = this.handleEnter.bind(this);
+    this.handleEntering = this.handleEntering.bind(this);
+    this.handleEntered = this.handleEntered.bind(this);
+    this.handleExit = this.handleExit.bind(this);
+    this.handleExiting = this.handleExiting.bind(this);
+  }
+
+  /* -- Expanding -- */
+  handleEnter(elem) {
+    const dimension = this._dimension();
+    elem.style[dimension] = '0';
+  }
+
+  handleEntering(elem) {
+    const dimension = this._dimension();
+    elem.style[dimension] = this._getScrollDimensionValue(elem, dimension);
+  }
+
+  handleEntered(elem) {
+    const dimension = this._dimension();
+    elem.style[dimension] = null;
+  }
+
+  /* -- Collapsing -- */
+  handleExit(elem) {
+    const dimension = this._dimension();
+    elem.style[dimension] = this.props.getDimensionValue(dimension, elem) + 'px';
+  }
+
+  handleExiting(elem) {
+    const dimension = this._dimension();
+
+    triggerBrowserReflow(elem);
+    elem.style[dimension] = '0';
+  }
+
+  _dimension() {
+    return typeof this.props.dimension === 'function'
+      ? this.props.dimension()
+      : this.props.dimension;
+  }
+
+  // for testing
+  _getTransitionInstance() {
+    return this.refs.transition;
+  }
+
+  _getScrollDimensionValue(elem, dimension) {
+    return `${elem[`scroll${capitalize(dimension)}`]}px`;
+  }
+
+  render() {
+    const {
+      onEnter, onEntering, onEntered, onExit, onExiting, className, ...props,
+    } = this.props;
+
+    delete props.dimension;
+    delete props.getDimensionValue;
```

Comment:
I think it makes more sense to delete props that we aren't using – it's more explicit and doesn't leave around an unused binding.